### PR TITLE
remove testZip step for selendroid

### PR DESCRIPTION
**Pull request by aluedeke**
fix for #4414

### CODE REVIEW

File: lib/helpers.js [hunk 1]
```diff
@@ -41,70 +41,28 @@ exports.copyLocalZip = function (localZipPath, cb) {
...
-  logger.debug("Testing zip archive: " + zipPath);
-  if (exports.isWindows()) {
-    if (fs.existsSync(zipPath)) {
-      logger.debug("Zip archive tested clean");
-      cb(null, true);
+    if (exports.isWindows()) {
+      var zip = new AdmZip(zipPath);
+      zip.extractAllTo(path.dirname(zipPath), true);
+      logger.debug("Unzip successful");
+      cb(null, null);
```
Comment (jlipps) [inline lib/helpers.js]: same thing here: `cb();`

File: lib/helpers.js [hunk 2]
```diff
@@ -41,70 +41,28 @@ exports.copyLocalZip = function (localZipPath, cb) {
...
+          logger.debug("Unzip successful");
+          cb(null, stderr);
         } else {
-          logger.error("Zip file " + zipPath + " was not valid");
+          logger.error("Unzip threw error " + err);
           logger.error("Stderr: " + stderr);
           logger.error("Stdout: " + stdout);
-          cb("Zip archive did not test successfully, check appium server logs " +
-             "for output", false);
+          cb("Archive could not be unzipped, check appium logs.", null);
```
Comment (jlipps) [inline lib/helpers.js]: wait, why is all this code duplicated between appium and appium-adb? it should definitely not be.

Comment (aluedeke) [inline lib/helpers.js]: confused me as well but had no time to investigate deeper. when using selendroid the code in appium-adb is called (if i remember correctly)

**Comment by jlipps** [comment]
Sorry @aluedeke but with a decent-sized change like this I think the right thing to do is refactor this out, probably so that it's in a separate package entirely (that can be used by appium-adb as well as appium). I really don't like having this duplicated code, and I'm not sure how it got that way.

**Comment by paymand** [comment]
I don't think this is necessary anymore. The latest unzip version on Ubuntu (6.0-12ubuntu1.3) has the fix and testZip step doesn't fail anymore.

**Pull request closed without merging**